pkg/upload/unity-android.go

```diff
@@ -0,0 +1,162 @@
+}
+
+func ProcessUnityAndroid(apiKey string, aabPath string, applicationId string, versionCode string, buildUuid string, arch string, versionName string, projectRoot string, paths []string, endpoint string, failOnUploadError bool, retries int, timeout int, overwrite bool, dryRun bool) error {
+	var err error
+	var zipPath string
+	var archList []string
+	var symbolFileList []string
+	var manifestData map[string]string
+	var aabManifestPath string
+
+	for _, path := range paths {
+		if utils.IsDir(path) {
+			zipPath, err = utils.FindFileWithSuffix(path, ".symbols.zip")
+
+			if err != nil {
+				return err
+			}
+
+			if aabPath == "" {
+				aabPath, err = utils.FindFileWithSuffix(path, ".aab")
+
+				if err != nil {
+					return err
+				}
+			}
+		} else if strings.HasSuffix(path, ".symbols.zip") {
+			zipPath = path
+
+			if aabPath == "" {
+				buildDirectory := filepath.Dir(path)
+
+				aabPath, err = utils.FindFileWithSuffix(buildDirectory, ".aab")
+
+				if err != nil {
+					return err
+				}
+			}
+		} else {
+			return fmt.Errorf("unsupported files. Please specify the unity `symbols.zip` file.")
+		}
+	}
+
+	log.Info("Using " + aabPath + " as the Unity Android AAB file")
+
+	log.Info("Using " + zipPath + " as the Unity Android symbols zip file")
+
+	aabDir, err := utils.ExtractFile(aabPath, "aab")
+
+	if err != nil {
+		return err
+	}
+
+	defer os.RemoveAll(aabDir)
+
+	aabManifestPathExpected := filepath.Join(aabDir, "base", "manifest", "AndroidManifest.xml")
+	if utils.FileExists(aabManifestPathExpected) {
+		aabManifestPath = aabManifestPathExpected
+	} else {
+		log.Warn("AndroidManifest.xml not found in AAB file")
+	}
+
+	if aabManifestPath != "" && (applicationId == "" || buildUuid == "" || versionCode == "" || versionName == "") {
+		log.Info("Reading data from AndroidManifest.xml")
+
+		manifestData, err = android.ProcessAabUploadOptions(aabManifestPath, apiKey, applicationId, buildUuid, versionCode, versionName)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	err = ProcessAndroidAab(manifestData["apiKey"], manifestData["applicationId"], manifestData["buildUuid"], []string{aabDir}, projectRoot, manifestData["versionCode"], manifestData["versionName"], endpoint, failOnUploadError, retries, timeout, overwrite, dryRun)
+
+	if err != nil {
+		return err
+	}
+
+	unityDir, err := utils.ExtractFile(zipPath, "unity-android")
+
+	if err != nil {
+		return err
+	}
+
+	defer os.RemoveAll(unityDir)
+
+	if arch == "" {
+		archList, err = utils.BuildFolderList([]string{unityDir})
+		if err != nil {
+			return err
+		}
+	} else {
+		archList = []string{arch}
+	}
+
+	for _, arch := range archList {
+		soPath := filepath.Join(unityDir, arch)
+		fileList, err := utils.BuildFileList([]string{soPath})
+		if err != nil {
+			return err
+		}
+		for _, file := range fileList {
+			if filepath.Base(file) == "libil2cpp.sym.so" && utils.ContainsString(fileList, "libil2cpp.dbg.so") {
```

Comment:
This definitely warrants some commentary:

> If both `.sym.so` and `.dbg.so` files are generated, favor the `.dbg.so`.

@richardelms - this might be a theoretical point, but would we do this for all files not just `il2cpp`? Would rather not have hard-coded file names in if we can help it.